modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/AbstractExtensionObjectFactory.java

```diff
@@ -134,6 +135,12 @@ protected ResolverSet getParametersAsResolverSet(ParameterizedModel model, List<
 
       if (resolver != null) {
         resolverSet.add(parameterName, resolver);
+      } else if (p.isRequired()) {
+        throw new IllegalStateException(format("Parameter '%s' of type %s from the %s '%s' is required but wasn't set",
+                                               parameterName,
+                                               getTypeId(p.getType()).orElse(StringUtils.EMPTY),
```

Comment:
use `ExtensionMetadataTypeUtils.getId`, it already handles the optional and a fallback for Java format. Maybe improve it to receive a default instead on it being always `""`